x-pack/legacy/plugins/apm/server/routes/default_api_types.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
+
+const tDate = new t.Type<string, string, unknown>(
+  'DateAsString',
+  (u): u is string => typeof u === 'string',
+  (u, c) =>
+    either.chain(t.string.validate(u, c), s => {
+      const d = new Date(s);
+      return isNaN(d.getTime()) ? t.failure(u, c) : t.success(s);
+    }),
+  a => a
+);
+
+export const defaultApiTypes = t.intersection([
```

Comment:
Perhaps a more correct name would be `defaultQueryParamTypes`?
Btw. I never really liked having default query param validations so if it makes sense to you I wouldn't mind ditching them and duplicating them everywhere. Only exception is perhaps `_debug` which should be built-in to the API/router.

Reply:
> Only exception is perhaps _debug which should be built-in to the API/router.

I think that should already be the case: even for APIs that don't have the default query params (time range and filter) they should still accept `_debug`

Reply:
I'll look into this, my feeling is that it's better to at least store them in a variable, because A) it's easy to forget one (although I think TS will catch that) B) `io-ts` types are not super readable, and every route would have a pretty long list of types. But I'll write one out to see how it looks.